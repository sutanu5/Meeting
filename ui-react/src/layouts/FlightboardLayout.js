import React, { Component } from 'react';

import Flightboard from '../components/flightboard/Flightboard';
import Navbar from '../components/flightboard/Navbar';

class FlightboardLayout extends Component {
  constructor(props) {
    super(props);
    this.state = {
      filter: ''
    };

    this.handleFilter = this.handleFilter.bind(this);
  }

  handleFilter(filterValue) {
    this.setState({
      filter: filterValue
    });
  }

  render () {
    const { filter } = this.state;
    return (
      <div id="page-wrap">
        <Navbar filter={this.handleFilter}/>
        <Flightboard filter={filter}/>
      </div>
    )
  }
}

export default FlightboardLayout;
