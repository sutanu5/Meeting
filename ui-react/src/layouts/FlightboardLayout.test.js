import FlightboardLayout from './FlightboardLayout';

describe('FlightboardLayout Component', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<FlightboardLayout />);

    expect(wrapper).toMatchSnapshot();
  });

  it('renders Navbar component', () => {
    const wrapper = shallow(<FlightboardLayout />);

    const component = wrapper.find('Navbar').exists();
    expect(component).toBeTruthy();
  });

  it('renders Navbar component props', () => {
    const wrapper = shallow(<FlightboardLayout />);

    const component = wrapper.find('Navbar').props().filter;
    expect(component).toBeTruthy();
  });

  it('calls handleFilter and sets state', () => {
    const wrapper = shallow(<FlightboardLayout />);

    let mockFn = jest.fn();
    FlightboardLayout.prototype.handleFilter = mockFn;

    const instance = wrapper.instance();
    const handleFilter = jest.spyOn(instance, 'handleFilter');
    
    instance.handleFilter('foo');

    expect(wrapper.state('filter')).toBe('foo');
    expect(handleFilter).toHaveBeenCalledWith('foo');
  });

  it('renders Flightboard component', () => {
    const wrapper = shallow(<FlightboardLayout />);

    const component = wrapper.find('Flightboard').exists();
    expect(component).toBeTruthy();
  });

  it('renders Flightboard component filter prop based on state', () => {
    const wrapper = shallow(<FlightboardLayout />);
    wrapper.setState({ filter: 'foo' });

    const component = wrapper.find('Flightboard').props().filter;
    expect(component).toBeTruthy();
  });
});
