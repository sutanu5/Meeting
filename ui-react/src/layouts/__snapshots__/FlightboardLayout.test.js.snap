// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlightboardLayout Component renders correctly 1`] = `
<div
  id="page-wrap"
>
  <Navbar
    filter={[Function]}
  />
  <Flightboard
    filter=""
  />
</div>
`;
